pkg/lib/authenticationflow/accept.go

```diff
@@ -119,20 +119,11 @@ func accept(ctx context.Context, deps *Dependencies, flows Flows, inputFn func(i
 		if errors.As(err, &errBotProtectionVerification) {
 			switch errBotProtectionVerification.Status {
 			case ErrorBotProtectionVerificationStatusSuccess:
-				uw, isJoinedError := err.(interface{ Unwrap() []error })
-				if isJoinedError {
-					errs := uw.Unwrap()
-					// make err become first non-ErrBotProtectionVerificationSuccess error
+				_, notMatched := errorutil.Partition(err, func(err error) bool {
 					var _errBPV *ErrorBotProtectionVerification
-					for _, _err := range errs {
-						if !errors.As(_err, &_errBPV) {
-							err = _err
-							break
-						}
-					}
-				} else {
-					err = nil
-				}
+					return errors.As(err, &_errBPV) && _errBPV.Status == ErrorBotProtectionVerificationStatusSuccess
+				})
+				err = notMatched
```

Comment:
line 122 to 126 should be applied to all 3 cases, not just the the success case. The rationale is that we want to extract the special error from the error chain, and set err to the rest of the error chain.